src/v0/destinations/webhook/utils.js

```diff
@@ -1,6 +1,37 @@
+const { isFeatureEnabled } = require('feature-flag-sdk');
+const { getXMLPayload } = require('../../../cdk/v2/destinations/http/utils');
 const { EventType } = require('../../../constants');
 const { getFieldValueFromMessage, flattenJson } = require('../../util');
 
+const JSON = 'JSON';
+const XML = 'XML';
+const FORM = 'FORM';
+
+const ContentTypeConstants = {
+  'application/json': JSON,
+  'application/xml': XML,
+  'text/xml': XML,
+  'application/x-www-form-urlencoded': FORM,
+};
+
+const getFormattedPayload = (headers, payload) => {
+  const normalizedHeaders = Object.keys(headers).reduce((acc, key) => {
+    acc[key.toLowerCase()] = headers[key];
+    return acc;
+  }, {});
+  const contentType = normalizedHeaders['content-type'];
+  const contentTypeSimplified = ContentTypeConstants[contentType] || JSON;
+
+  switch (contentTypeSimplified) {
+    case XML:
+      return { payload: { payload: getXMLPayload(payload) }, contentTypeSimplified };
+    case FORM:
+      return { payload, contentTypeSimplified };
```

Comment:
```suggestion
      return { getFORMPayload(payload), contentTypeSimplified };
```

Reply:
We don't have to convert this into FORMPayload ourselves as `rudder-server` takes care of it.
Follow this link to see what rudder-server is doing: https://github.com/rudderlabs/rudder-server/blob/f22e8f4bb61893af132f3f4386155c83c695fd51/router/network.go#L124

I've left it as switch case to make it readable.